### ISSUE TITLE
Bundler connection to rubygems.org failed; Ruby & RubyGems connection succeeded

### ISSUE THREAD

**Issue by EvanGrillo** [bundler/bundler#6681]
## input
```
ruby -ropen-uri -e 'eval open("https://git.io/vQhWq").read'
```
## output
```
Here's your Ruby and OpenSSL environment:

Ruby:           2.5.1p57 (2018-03-29 revision 63029) [x86_64-darwin16]
RubyGems:       2.7.7
Bundler:        1.16.2
Compiled with:  OpenSSL 1.0.2p  14 Aug 2018
Loaded version: OpenSSL 1.0.2p  14 Aug 2018
SSL_CERT_FILE:  /usr/local/etc/openssl/cert.pem
SSL_CERT_DIR:   /usr/local/etc/openssl/certs

With that out of the way, let's see if you can connect to rubygems.org...

Bundler connection to rubygems.org:       failed  ❌  (execution expired)
RubyGems connection to rubygems.org:      success ✅
Ruby net/http connection to rubygems.org: success ✅

Although your Ruby installation and RubyGems can both connect to rubygems.org, Bundler is having trouble. The most likely way to fix this is to upgrade Bundler by running `gem install bundler`. Run this script again after doing that to make sure everything is all set. If you're still having trouble, check out the troubleshooting guide at http://ruby.to/ssl-check-failed 📦
(eval):136: warning: constant OpenSSL::SSL::SSLContext::METHODS is deprecated
```

**Comment by indirect** [comment]
It's not likely to help, but you could try `gem install bundler` to get the very latest Bundler before trying again. In my experience, "execution expired" errors typically mean something has gone wrong at the OS or network level... packet loss, connection dropped, firewall somewhere in the way, or something like that.

Please try following [ISSUES](https://github.com/bundler/bundler/blob/master/doc/contributing/ISSUES.md#filing-issues-a-guide) and reporting back with the results and the various information requested in that doc.

**Comment by EvanGrillo** [comment]
### could the issue be in here?
```
Compiled with:  OpenSSL 1.0.2p  14 Aug 2018
Loaded version: OpenSSL 1.0.2p  14 Aug 2018
SSL_CERT_FILE:  /usr/local/etc/openssl/cert.pem
SSL_CERT_DIR:   /usr/local/etc/openssl/certs
```

### as instructed by Bundler, I added a .pem

<img width="772" alt="screen shot 2018-08-29 at 11 29 07 pm" src="https://user-images.githubusercontent.com/32852965/44829648-6417b000-abe3-11e8-9abf-8c4fd4cb027c.png">

**Comment by EvanGrillo** [comment]
Bundle install now works unless I add new gems to the Gemfile without first manually installing them. Deployment to Heroku is also disrupted:
```
remote:  !     Failed to install gems via Bundler.
```

**Comment by sgtcoolguy** [comment]
Just a note that I'm having this exact issue with a number of CI build machines using chef 14.x with it's embedded ruby/gems/bundler.

I had to actually manually change those boxes to bump the timeout to 60 as they're just extremely slow using bundler and timing out. (`chef exec bundle config timeout 60`).

Anyways, in looking at which boxes were affect and which were not, it seems that the slow boxes are all macOS 10.12.x, while the "good" boxes are all macOS 10.13.x.

**Comment by EvanGrillo** [comment]
@sgtcoolguy thanks for sharing. I'm using Sierra 10.12.6. Would you recommend that I run an update or try anything to resolve the connection error?

**Comment by sgtcoolguy** [comment]
@EvanGrillo Personally I just bumped up the timeout since this was only about getting chef to install some gems bundled with a couple cookbooks, so we're not using bundler heavily.

If you can upgrade a box from 10.12 to 10.13 it'd certainly be interesting to see if the issue disappeared. I may try that with one of them in the next week or so.

I don't have physical access to the boxes to try and make a determination if it's just a coincidence that these boxes are all 10.12.6, or if they're on the same network switch or what. I've asked someone with access to take a look.

For more background, these are all various vintage Mac mini's used as CI nodes under Jenkins and managed by Chef. It's in an office network and behind a firewall. They're all using the same gateway and DNS servers. Most have 100baseT and a few have 1000baseT ethernet connections. The only noticeable difference that I can tell is that the ones timing out are on 10.12.6 while the working ones are on 10.13.6. But again, I have asked someone in IT to take a closer look as I'm not 100% convinced that it's the OS version causing this bad combination.

**Comment by EvanGrillo** [comment]
@sgtcoolguy @indirect I update MacOS High Sierra to version 10.13.6 and the Bundler connection to rubygems.org is now green 
<img width="1056" alt="screen shot 2018-09-06 at 7 41 10 pm" src="https://user-images.githubusercontent.com/32852965/45192250-df451b80-b20c-11e8-9149-2d21ddabc466.png">

@sgtcoolguy thanks for your input!

**Comment by EvanGrillo** [comment]
@sgtcoolguy I agree 100% because as pointed out in creating the ticket, it's an isolated connection error

**Comment by sudo-eddy** [comment]
I'm having the same issue, but running Ubuntu 16.04 Kernel version 4.15, I wonder if I should update my OS as well? hmm
